cargo-fix/tests/all/broken_build.rs

```diff
@@ -111,6 +114,7 @@ fn broken_fixes_backed_out() {
 
     // Attempt to fix code, but our shim will always fail the second compile
     p.expect_cmd("cargo-fix fix")
+        .env("__CARGO_FIX_YOLO", "true")
```

Comment:
Hm does this mean that the lints we're testing against aren't currently classifed as "machine applicable"?

Reply:
Yeah

Reply:
Hm ok, could we perhaps start testing lints that are classified as machine applicable?

Reply:
Well, like I said, we don't have many -- just unreachable_pub, bare_trait_object, and abs_path_with_module. Do you want to rewrite the tests to always use those? I'd wait a bit for more applicability markers to land and then refactor the tests.

Reply:
Hm ok, the 2018 compatibility lints are at least machine applicable, right? If so can you make sure that they're already tested and/or add a test for that?

Reply:
Right -- everything but the `upgrade_extern_crate` case work without the env var. Updated.

Reply:
Nice!

As one final comment, could this be an extension method for just the test suite to do something like `.fix_everything()` instead of duplicating the env var? Other than that r=me!

Reply:
Good idea!